kaitai_utils.py

```diff
@@ -37,16 +39,66 @@ def get_plugin_title_from_metadata():
         return _to_camel_case(metadata["id"])
 
 
-def write_to_json(data_binary: BinaryIO, writer: BufferedWriter, class_type: Type[KaitaiStruct]):
-    """
-    Writes a binary form of JSON string into a BufferedWriter
-
-    @param data_binary: binary data containing the file content
-    @param writer: bufferedWriter to write the binary form of the JSON string to
-    @param class_type: class that contains the parsing to a KaiTai struct
-    @return: JSON string representing contents of data object
-    """
-    writer.write(bytes(to_json_string(data_binary, class_type), "utf-8"))
+class JsonWriter:
```

Comment:
perhaps 
```suggestion
class KaitaiToTraceWriter:
```
(given that this does more than writing JSON now, but also adds child-traces)

Reply:
done